presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -367,7 +362,12 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                     rightSource = new ProjectNode(idAllocator.getNextId(), rightSource, rightProjections.build());
                     criteria = builder.build();
                 }
-                output = new JoinNode(node.getId(), node.getType(), leftSource, rightSource, criteria, node.getLeftHashSymbol(), node.getRightHashSymbol());
+                if (rewriteToCrossJoin) {
```

Comment:
I don't think you need to special check for cross join above. just do the check here to see if criteria is empty before deciding to do cross join.